mlx/linalg.cpp

```diff
@@ -78,13 +87,13 @@ inline array matrix_norm(
     bool keepdims,
     StreamOrDevice s) {
   if (ord == "f" || ord == "fro") {
-    return sqrt(sum(abs(a, s) * abs(a, s), axis, keepdims, s));
+    return sqrt(sum(square(a, s), axis, keepdims, s), s);
```

Comment:
If we want to support complex numbers, we might want to replace `square(a,s)` with `a times conjugate(a)`.

Reply:
It's a good point, I didn't think of that. We should probably modify to branch on complex types and do the abs * abs thing for now if complex (we don't have conjugate yet).

Reply:
I implemented branching on complex types.